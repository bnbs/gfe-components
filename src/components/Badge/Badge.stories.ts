import { Meta, StoryObj } from "@storybook/react/*";

import Badge from "./Badge";

const meta = {
  component: Badge,
  title: "Badge",
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof Badge>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: "test",
    color: "neutral",
    size: "sm",
  },
};
