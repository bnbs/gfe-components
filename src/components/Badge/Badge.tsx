import React, {
  ComponentProps,
  ForwardedRef,
  forwardRef,
  ReactNode,
} from "react";
import clsx from "clsx";

const colorClasses = {
  neutral: "bg-gray-50 text-neutral-600 border-neutral-200",
  error: "bg-red-50 text-red-600 border-red-200",
  warning: "bg-amber-50 text-amber-700 border-amber-200",
  success: "bg-green-50 text-green-700 border-green-200",
  brand: "bg-indigo-50 text-indigo-700 border-indigo-200",
};

const sizeClasses = {
  sm: "text-xs px-1.5 py-0.5",
  md: "text-sm px-2 py-0.5",
  lg: "text-sm px-2.5 py-1",
};

export interface BadgeProps extends ComponentProps<"span"> {
  /** @default sm */
  size?: "sm" | "md" | "lg";
  /** @default neutral */
  color?: "neutral" | "error" | "warning" | "success" | "brand";
  children: ReactNode;
}

const Badge = forwardRef<HTMLSpanElement, BadgeProps>(
  (props: BadgeProps, ref: ForwardedRef<HTMLSpanElement>) => {
    const { children, color = "neutral", size = "sm", ...restProps } = props;

    return (
      <span
        ref={ref}
        className={clsx(
          "rounded-full border border-solid w-fit",
          sizeClasses[size],
          colorClasses[color]
        )}
        {...restProps}
      >
        {children}
      </span>
    );
  }
);

export default Badge;
