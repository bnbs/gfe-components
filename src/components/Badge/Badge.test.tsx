import React from "react";
import { render, screen } from "@testing-library/react";

import Badge, { BadgeProps } from "./Badge";

const defaultProps: BadgeProps = {
  children: "This is a Badge",
  color: "neutral",
  size: "sm",
};

describe("Badge", () => {
  test.each(["sm", "md", "lg"] as BadgeProps["size"][])(
    "renders the Badge with size equal to %s",
    (size) => {
      const component = render(<Badge {...defaultProps} size={size} />);
      expect(component).toMatchSnapshot();
    }
  );

  test.each([
    "neutral",
    "error",
    "warning",
    "success",
    "brand",
  ] as BadgeProps["color"][])(
    "renders the Badge with color equal to %s",
    (color) => {
      const component = render(<Badge {...defaultProps} color={color} />);
      expect(component).toMatchSnapshot();
    }
  );

  test("renders the Badge with children equal to a react element", () => {
    const text = "This is a react element";
    render(
      <Badge {...defaultProps}>
        <div>{text}</div>
      </Badge>
    );

    expect(screen.getByText(text)).toBeDefined();
  });
});
