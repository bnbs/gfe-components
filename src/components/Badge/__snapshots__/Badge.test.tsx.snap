// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders the Badge with color equal to brand 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-indigo-50 text-indigo-700 border-indigo-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-indigo-50 text-indigo-700 border-indigo-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with color equal to error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-red-50 text-red-600 border-red-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-red-50 text-red-600 border-red-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with color equal to neutral 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with color equal to success 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-green-50 text-green-700 border-green-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-green-50 text-green-700 border-green-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with color equal to warning 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-amber-50 text-amber-700 border-amber-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-amber-50 text-amber-700 border-amber-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with size equal to lg 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-sm px-2.5 py-1 bg-gray-50 text-neutral-600 border-neutral-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-sm px-2.5 py-1 bg-gray-50 text-neutral-600 border-neutral-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with size equal to md 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-sm px-2 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-sm px-2 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders the Badge with size equal to sm 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
      >
        This is a Badge
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded-full border border-solid w-fit text-xs px-1.5 py-0.5 bg-gray-50 text-neutral-600 border-neutral-200"
    >
      This is a Badge
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
